rfcs/xxxx-template.md

```diff
@@ -0,0 +1,42 @@
+- Start Date: (2023-04-18)
+- RFC PR:
+
+# Summary
+Implement a category dropdown in the admin UI for both Collection & Single types.
+
+# Motivation
+The motivation behind this proposal is to simplify content organization and improve the user experience while managing content in the Strapi admin interface. Currently, users have to navigate through a long list of Collection and Single types, which can be time-consuming and cumbersome, especially when dealing with numerous content types. By introducing a category dropdown, users can easily group and filter their content types, making content management more efficient and user-friendly.
+
+# Detailed design
+The proposed solution includes the following design changes:
+
+1. Add a category dropdown in the admin UI for both Collection & Single types, enabling users to filter content types based on the selected category.
+2. Allow users to create, edit, and delete categories from the admin UI.
+2. Automatically assign newly created content types to a default category (e.g., "Uncategorized") if no category is selected.
+2. Enable users to change a content type's category through the content type's settings.
+2. Display the category alongside the content type in the main admin UI.
+
+Edge Cases:
+
+- Handling empty categories: If all content types within a category are deleted or moved to another category, the empty category should still be displayed in the dropdown to allow users to assign new content types to it.
+- Handling category deletion: If a category is deleted, all associated content types should be moved to the default category.
+
+# Example
+No changes to the current API are required for this proposal. The proposed changes are only related to the admin UI.
```

Comment:
Same place the category info for components is saved. 
I believe it's on file, not in DB. Looked into the core code with Bogie and it doesn't seem like a huge change overall.

Reply:
saving it in the file is the best option this will mean that the server would have to be restarted every time someone changes that 
I would do 1 stap back with the proposal since we can't change the folder stucture for this I think it will be better to not save the categories groups in the file system but add a hook inside the content-type/single-type loading system and do that part in memory or not at all

Reply:
> Where would you like to store this category information?

As these categories improve the user experience in the Admin panel, saving it **in the database** is a better solution. A category as proposed now has no effect on the model structure or other backend implications. This way you also allow an easier way to customize the behavior of categories with their order & translations.


_(And maybe if you have a good config structure this could also allow an option to save a user-personalized order, which could be saved in session for example.)_

Reply:
I  think  first version  should  be file  based  and then later we can add db on top since now for every new strapi instance with a fresh db you now to redo all your categories

Reply:
Well, I don't like to see the schema file 'polluted' with Admin panel labels.
> with a fresh db you now to redo all your categories since they are all only saved in the db

That's why we have [config-sync](https://market.strapi.io/plugins/strapi-plugin-config-sync) and hopefully soon a **selective** data transfer system.

Don't get me wrong, I love to configure settings in file/git, but I see a better fit for these categories/labels in the database, which will also allow for **order** & **internationalization**.

Reply:
@MattieBelt Only thing it would add is ordering since internationalization would not be possible on the db layer without i18n installed and we can't assume it is. Also this would be a core strapi feature so if posible no plugins should be needed to make it work properly.

Reply:
Yes, it is possible to add translations and build it upon the translation/language system that exists in the core admin panel. 
The database saved config could follow a similar naming system as the translation files, with `useLocalesProvider` and maybe even `formatMessage` it can be built in. 

(I'm sorry translations would have been a better word choice, I edited my comment above. I also deliberately didn't use the term 'i18n' as that would point to the plugin, but that isn't the case 😉)